### ISSUE TITLE
virtual repo redirected to ext repo fails, when ext repo has rpm packages under 2 or more nested directories

### ISSUE THREAD

**Issue by liangderong1992** [ImmobilienScout24/yum-repo-server#63]
I have a repository which has couple sub-directories under it and I would like to create a virtual repo redirecting to my own repo. 

For example, my own repo under http://repo.myownrepo.com/MyOwnRepo is:
MyOwnRepo
----DirectoryA
  ---- DirectoryB
    ---- DirectoryC
      ---- sample.rpm

On the system running yum-repo-server, I set up a virtual repo MyVirtualRepo and redirected it to http://repo.myownrepo.com/MyOwnRepo

When I use the link http://[yum-repo-server-ip]:[port]/yum-repo-server/repo/virtual/MyVirtualRepo/DirectoryA/DirectoryB, it returns a page showing DirectoryC.

But when I use the link  http://[yum-repo-server-ip]:[port]/yum-repo-server/repo/virtual/MyVirtualRepo/DirectoryA/DirectoryB/DirectoryC, it returns status 404.

Same status 404 for http://[yum-repo-server-ip]:[port]/yum-repo-server/repo/virtual/MyVirtualRepo/DirectoryA/DirectoryB/DirectoryC/sample.rpm.

By the way, http://repo.myownrepo.com/MyOwnRepo/DirectoryA/DirectoryB/DirectoryC/sample.rpm works fine to get the package.

**Comment by oli99sc** [comment]
yes you are right, virtual repo controler matches against /virtual/<repo>/<arch>/<file name> pattern.
I am currently looking for a solution.
